web/packages/shared/utils/processRedirectUri.ts

```diff
@@ -0,0 +1,72 @@
+/**
+ * Teleport
+ * Copyright (C) 2024 Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/**
+ * Processes a redirect URI to ensure it's valid and follows the expected format.
+ *
+ * This function handles various cases:
+ * - Null or empty input: Returns the basePath
+ * - Full URLs:
+ *   - External: Uses only the pathname, prepending the basePath if not already present
+ *   - Internal: Prepends the basePath if not already present
+ * - Relative paths: Prepends the basePath if not already present
+ * - Invalid URIs: Returns the basePath
+ *
+ * @param basePath - The base path to prepend to the processed URI (e.g., '/web')
+ * @param redirectURI - The redirect URI to process. Can be null, a relative path, or a full URL.
+ * @returns A processed URI string that always starts with the basePath, unless it's an invalid input.
+ *
+ * @example
+ * processRedirectURI('/web', null) // returns '/web'
+ * processRedirectURI('/web', 'https://example.com/path') // returns '/web/path'
+ * processRedirectURI('/web', '/custom/path') // returns '/web/custom/path'
+ * processRedirectURI('/web', '/web/existing/path') // returns '/web/existing/path'
+ * processRedirectURI('/web', 'invalid://url') // returns '/web'
+ * processRedirectURI('/app', 'https://example.com/path') // returns '/app/path'
+ */
+export function processRedirectURI(
+  basePath: string,
```

Comment:
the purpose of adding in the `basePath` here was so I could pass in something like `cfg.routes.root` and if that ever changed, this would be updated with it. However, I wasn't able to access that `cfg` everywhere this is used and had to manually pass `"/web"` anyway so... I could be convinced to just remove it and hard code in the `"/web"` . I really dont see that changing anytime anyway.

Reply:
I guess we can hardcode it? I'd only leave a comment that the value is equal to `cfg.routes.root`.